packages/core/src/render3/instructions/styling.ts

```diff
@@ -302,11 +305,12 @@ export function classMapInternal(
   // there should not be a situation where a directive host bindings function
   // evaluates the inputs (this should only happen in the template function)
   if (!isHostStyling() && hasDirectiveInput && classes !== NO_CHANGE) {
-    updateDirectiveInputValue(context, lView, tNode, bindingIndex, classes, true);
+    updateDirectiveInputValue(
+        context, lView, tNode, bindingIndex, classes, true, tView.firstUpdatePass);
     classes = NO_CHANGE;
   }
 
-  stylingMap(context, tNode, lView, bindingIndex, classes, true, hasDirectiveInput);
+  stylingMap(context, tNode, tView, lView, bindingIndex, classes, true, hasDirectiveInput);
```

Comment:
Why not pass through `tView.firstUpdatePass`  at this level? That way, we can avoid reading the property twice (once in line 309 and once in `stylingMap`).
```suggestion
  stylingMap(context, tNode, firstUpdatePass, lView, bindingIndex, classes, true, hasDirectiveInput);
```

Reply:
fixed